Address gettext on primary and secondary navigation items

AparnaKarve opened the pull request:
Navigation text and tooltip for primary and secondary navigation items shows up in English on non `en` locales.

PR https://github.com/ManageIQ/manageiq-ui-service/pull/277 probably meant to resolve this issue, but it did not address it completely.

https://bugzilla.redhat.com/show_bug.cgi?id=1389655

Before:
<img width="1385" alt="screen shot 2016-10-28 at 4 39 36 pm" src="https://cloud.githubusercontent.com/assets/1538216/19825374/2c0ca600-9d2d-11e6-8387-da192aced80c.png">

After:
<img width="1382" alt="screen shot 2016-10-28 at 4 39 11 pm" src="https://cloud.githubusercontent.com/assets/1538216/19825379/394e79d8-9d2d-11e6-8382-f718c2a0feee.png">

AparnaKarve commented:
@himdel @mzazrivec Review please.

himdel commented:
@AparnaKarve the change looks good, but I'd move it to `client/app/config/navigation.config.js`, in `createItem`. That way, you won't have to repeat twice for primary/secondary..

chriskacerguis commented:
I'm thinking that the circles should align (See screenshot), instead of just centered.

<img width="912" alt="screen shot 2016-10-29 at 8 22 35 am" src="https://cloud.githubusercontent.com/assets/2414647/19829829/f6467c92-9db0-11e6-958f-0542d643ae98.png">

chriskacerguis commented:
@AparnaKarve disregard last comment.

mzazrivec commented:
@AparnaKarve Yes, it works. Although I'd probably also do the thing @himdel mentioned
in his comment.

AparnaKarve commented:
> but I'd move it to client/app/config/navigation.config.js, in createItem. That way, you won't have to repeat twice for primary/secondary

@himdel @mzazrivec Personally I would've loved it to move it there too, but ...
 Houston, We've Got a problem <a href=https://en.wikipedia.org/wiki/Houston,_We%27ve_Got_a_Problem>™</a>.
![image](https://cloud.githubusercontent.com/assets/1538216/19866719/79fd90b2-9f5e-11e6-8249-e7e206b2eb1b.png)

Let me explain (and correct me if I'm wrong) --
Isn't `client/app/config/navigation.config.js` completely off limits for doing any sort of `gettext` activity like `__("SomeText")` ?
And that is probably the reason why `N_("SomeText")` was used <a href=https://github.com/ManageIQ/manageiq-ui-service/blob/master/client/app/config/navigation.config.js#L12>here</a> and other places to delay the gettext substitution until we get to the controller phase.

What I could probably do is DRY up the above logic so that both `primary` and `secondary` are handled in one common function. Thoughts?

chriskacerguis commented:
Adding WIP label.

mzazrivec commented:
@AparnaKarve Nah, if what I suggested doesn't work, then I'm fine with the current state. It could
be extracted into a separate function, but this is not a big deal really. Agreed?

chriskacerguis commented:
@AparnaKarve I have added the WIP label to ensure this does not get merged (there seems to be some active discussion on it), if you could please remove it when you are ready for me to review / merge I would appreciate it.

AparnaKarve commented:
@miq-bot remove_label wip

AparnaKarve commented:
@mzazrivec I had a hunch that the suggested solution would not work -- but wanted to give it the benefit of the doubt anyway. 
I'm OK with the current state as well, but FWIW, I've extracted the common code out in a separate function.

@chriskacerguis I've removed the WIP label. Let's get this merged soon since these strings in the Navigation bar are quite prominent and have a high visibility.

chriskacerguis commented:
LGTM.

AparnaKarve commented:
@chriskacerguis Adding the `euwe/yes` label for this...

@miq-bot add_label euwe/yes

chessbyte commented:
Euwe Backport details:
```sh
$ git log -1
commit 5a15590548661a94eed6739d191b03b22e41e5c2
Author: Chris Kacerguis <chriskacerguis@users.noreply.github.com>
Date:   Tue Nov 1 11:00:48 2016 -0500

    Merge pull request #299 from AparnaKarve/bz1389655_ssui_gettext_for_nav_text
    
    Address gettext on primary and secondary navigation items
    (cherry picked from commit 9aa0585a9aced7e91bbc359c72bef2b3a5b7df3c)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1389655
```

chessbyte commented:
@simaishi Please make Euwe BZ for https://bugzilla.redhat.com/show_bug.cgi?id=1389655

simaishi commented:
Euwe BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1391142
